example-app/app/Models/City.php

```diff
@@ -9,6 +9,23 @@ class City extends Model
 {
     use HasFactory;
 
+    protected $fillable = ['name'];
+
+    public function scopeFilter($query, array $filters)
+    {
+        $query->when($filters['airline_id'] ?? false, function ($query, $airlineId) {
+            $query->whereHas('departureFlights', function ($q) use ($airlineId) {
+                $q->where('airline_id', $airlineId);
+            })->orWhereHas('arrivalFlights', function ($q) use ($airlineId) {
```

Comment:
```suggestion
            $query->whereHas('departureFlights', fn ($q) => $q->where('airline_id', $airlineId))
            ->orWhereHas('arrivalFlights', function ($q) use ($airlineId) {
```

you can use fn short syntax for it if it's one line